File: deleteUsers/handler.py
import boto3, json, os

client = boto3.resource('dynamodb')

IS_OFFLINE = os.getenv('IS_OFFLINE')
if IS_OFFLINE:
    boto3.Session(
        aws_access_key_id="dummy",
        aws_secret_access_key="dummy",
    )
    client = boto3.resource('dynamodb', endpoint_url="http://localhost:8000")

table = client.Table('usersTable')


def deleteUsers(event, context):
    user_id = event['pathParameters']['id']

    result = table.delete_item(Key = {'pk': user_id})

    body = json.dumps({ 'message': f"User {user_id} deleted succesfully" })

    response = {
        "statusCode": result['ResponseMetadata']['HTTPStatusCode'],
        "body": body
    }

    return response

File: functions/updateUsers/handler.js
const aws = require('aws-sdk')

let dynamodbClientParams = {}

if (process.env.IS_OFFLINE) {
    dynamodbClientParams = {
        region: 'localhost',
        endpoint: 'http://localhost:8000',
        accessKeyId: 'dummy',
        secretAccessKey: 'dummy'
    }
}

const dynamodb = new aws.DynamoDB.DocumentClient(dynamodbClientParams)

const updateUsers = async (event, context) => {

    let userId = event.pathParameters.id;

    const body = JSON.parse(event.body);

    const params = {
        TableName: 'usersTable',
        Key: { pk: userId },
        UpdateExpression: 'set #name = :name, #edad = :edad, #email = :email, #telefono = :telefono',
        ExpressionAttributeNames: { '#name': 'name', '#edad': 'edad', '#email': 'email', '#telefono': 'telefono' },
        ExpressionAttributeValues: 
            {   ':name': body.name,
                ':edad': body.edad,
                ':email': body.email,
                ':telefono': body.telefono
             },
        ReturnValues: 'UPDATED_NEW'
    };

    const res = await dynamodb.update(params).promise();
    console.log(res);
    return {
        statusCode: 200,
        body: JSON.stringify({'User updated': res.Attributes }), 
    };

};

module.exports = {
    updateUsers
}
